Src/Support/GoogleApis.Tests/Apis/Download/MediaDownloaderTest.cs

```diff
@@ -293,6 +291,31 @@ public void DownloadAsync_Cancel()
             Subtest_Download_Chunks(MediaContent.Length - 1, false, 1);
         }
 
+        [Test]
+        public void DownloadRange()
+        {
+            using (var service = new MockClientService())
+            {
+                int startInclusive = 5;
+                int endInclusive = 13;
+                var downloader = new MediaDownloader(service);
+                downloader.Range = new RangeHeaderValue(startInclusive, endInclusive);
+                var outputStream = new MemoryStream();
+                var result = downloader.Download(_httpPrefix + "content", outputStream);
+
+                Assert.AreEqual(result.Status, DownloadStatus.Completed);
+                Assert.IsNull(result.Exception);
+                Assert.AreEqual(result.BytesDownloaded, outputStream.Position);
+
+                byte[] bytes = outputStream.ToArray();
+                Assert.AreEqual(endInclusive - startInclusive + 1, bytes.Length);
+                for (int i = 0; i < bytes.Length; i++)
+                {
+                    Assert.AreEqual(MediaContent[i + startInclusive], bytes[i]);

```

Comment:
Have added another commit for a different option. Not sure which I really prefer.